integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/DatabricksEnvironmentFacetBuilder.java

```diff
@@ -71,17 +74,53 @@ private Map<String, Object> getDatabricksEnvironmentalAttributes(SparkListenerJo
      * access it via reflection.
      */
     try {
-      dbutilsClass = Class.forName("com.databricks.dbutils_v1.impl.DbfsUtilsImpl");
-      dbutils = (DbfsUtils) dbutilsClass.getDeclaredConstructor().newInstance();
-      dbProperties.put("mountPoints", getDatabricksMountpoints(dbutils));
+      DbfsUtils dbfsUtils = getDbfsUtils();
+      if (dbfsUtils == null) {
+        dbProperties.put("mountPoints", new ArrayList<DatabricksMountpoint>());
+      } else {
+        dbProperties.put("mountPoints", getDatabricksMountpoints(dbfsUtils));
+      }
+
     } catch (Exception e) {
-      log.warn("Failed to load dbutils in OpenLineageListener");
+      log.warn("Failed to load dbutils in OpenLineageListener:", e);
       dbProperties.put("mountPoints", new ArrayList<DatabricksMountpoint>());
     }
 
     return dbProperties;
   }
 
+  // Starting in Databricks Runtime 11, there is a new constructor for DbFsUtils
+  // If running on an older version, the constructor has no parameters.
+  // If running on DBR 11 or above, you need to specify whether you allow mount operations (true or
+  // false)
+  private static DbfsUtils getDbfsUtils()
```

Comment:
Just to make sure: couldn't we use here `com.databricks.dbutils_v1.DBUtilsHolder.dbutils().fs()` instead of creating `DbfsUtils` on our own?

Reply:
@pawel-big-lebowski - Are you thinking of using the [DBUtils API](https://learn.microsoft.com/en-us/azure/databricks/dev-tools/databricks-utils#--databricks-utilities-api-library) if so, the pre-instantiated dbutils is not available to the spark listener.  We [had this challenge](https://stackoverflow.com/questions/69859733/how-to-use-dbutils-in-a-sparklistener-on-databricks) when first instantiating this class.